cpp/src/arrow/flight/server_tracing_middleware.cc

```diff
@@ -29,15 +29,31 @@
 #include <opentelemetry/context/propagation/global_propagator.h>
 #include <opentelemetry/context/propagation/text_map_propagator.h>
 #include <opentelemetry/trace/context.h>
-#include <opentelemetry/trace/experimental_semantic_conventions.h>
 #include <opentelemetry/trace/propagation/http_trace_context.h>
+#include <opentelemetry/trace/semantic_conventions.h>
 #endif
 
 namespace arrow {
 namespace flight {
 
 #ifdef ARROW_WITH_OPENTELEMETRY
 namespace otel = opentelemetry;
+
+// TODO: Update this once opentelemetry-cpp exposes minor version as a macro
+// https://github.com/open-telemetry/opentelemetry-cpp/issues/2012
+// TODO: Remove once we drop support for opentelemetry-cpp < 1.8.0
+// They switched from ALL_CAPS to kConstantFormat in 1.8.0. But we can't check
+// the minor version until they expose that. So, for now, we vendor these constants.
+namespace SemanticConventions {
+static constexpr const char* kRpcGrpcStatusCode = "rpc.grpc.status_code";
```

Comment:
nit: maybe const std::string_view to prefer new-style vs raw C constants?

Reply:
Their APIs don't accept `std::string_view` (only `nostd::string_view`).